### PR TITLE
Allow multiple featured documents.

### PR DESCRIPTION
**Pull request by krues8dr**
#825 This one got a lot more complicated than I expected.  @sethetter I backed off the featured routes from your new middleware on these since only Admins can do that and it's not document-based.

@sethetter @cmbirk

### CODE REVIEW

File: public/js/controllers/dashboardEditorController.js [hunk 1]
```diff
@@ -748,10 +750,15 @@ angular.module('madisonApp.controllers')
...
         });
+        */
       };
 
       $scope._setFeaturedDoc = function () {
         return $http.post('/api/docs/featured', {id: $scope.doc.id});
       };
+
+      $scope.removeFeaturedDoc = function() {
+        return $http.delete('/api/docs/featured/' + $scope.doc.id);
```
Comment (sethetter) [inline public/js/controllers/dashboardEditorController.js]: When this is run by the "Remove" button in the doc edit settings screen, it successfully hits the API route but doesn't update the state on the page to show it was successfully removed.

Comment (krues8dr) [inline public/js/controllers/dashboardEditorController.js]: Ah, good catch.  I'll have a look at this, thanks!

File: public/js/controllers/siteSettingsController.js [hunk 2]
```diff
@@ -78,4 +79,38 @@ angular.module('madisonApp.controllers')
...
+        },
+        function(error) {
+          console.log('error', error);
+          growl.error($translate.instant('errors.general.save'));
+        });
+    };
+
+    $scope.removeFeaturedDoc = function (doc) {
+      $http.delete('/api/docs/featured/'+doc.id).then(function (data) {
+        $scope.featuredDocs = data.data;
```
Comment (sethetter) [inline public/js/controllers/siteSettingsController.js]: Is it worthwhile to clarify that if the last featured doc is removed using this screen that a sensible default is chosen? Or maybe restrict the removal of a featured doc if there is only one so they have to pick a new one first?

Comment (krues8dr) [inline public/js/controllers/siteSettingsController.js]: I don't want to force the user to have to pick one, but yes we definitely should let them know what will happen!

File: public/js/controllers/dashboardEditorController.js (hunk 1 above)
Comment (sethetter) [inline public/js/controllers/dashboardEditorController.js]: Looks like same is needed for setting as a featured doc, FYI.

**Comment by krues8dr** [comment]
@sethetter  better now?

**Pull request merged**